src/Database/Type/DateTimeType.php

```diff
@@ -308,7 +308,6 @@ public function marshal(mixed $value): ?DateTimeInterface
             if ($value instanceof NativeDateTime) {
                 $value = clone $value;
             }
-            assert($value instanceof DateTime || $value instanceof DateTimeImmutable);
```

Comment:
This should be NativeDateTime || DateTimeImmutable. Removed since static analyers complain it's always true.